zmq4_routerdealer_test.go

```diff
@@ -239,3 +240,35 @@ func TestRouterDealer(t *testing.T) {
 		})
 	}
 }
+
+func TestRouterClose(t *testing.T) {
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+	router := zmq4.NewRouter(ctx)
+	err := router.Listen("tcp://*:*")
+	if err != nil {
+		t.Errorf("want error %v, got %q", nil, err)
```

Comment:
```suggestion
		t.Fatalf("router could not listen: %+v", err)
```

Reply:
No, no, #57 is a different issue, I looked in the code and if the router doesn't have any connections it returns an error on `router.Close()`, is that the intended behavior?

Reply:
yes, right you are.

to be honest, I don't recall exactly _why_ I put that in.
quickly re-reading through the specs, I don't think that's actually warranted.